File: leetcode/0001-1000/0301-0400/0334-increasing-triplet-subsequence/IncreasingTripletSubsequence.java
public class IncreasingTripletSubsequence {
  public static void main(String[] args) {
    System.out.println(new Solution().increasingTriplet(new int[] { 1, 2 }));
  }
}

class Solution {
  public boolean increasingTriplet(int[] numbers) {
    int smallest = Integer.MAX_VALUE;
    int middle = Integer.MAX_VALUE;

    for (int number : numbers) {
      if (number > middle) {
        return true;
      }

      if (number <= smallest) {
        smallest = number;
      } else {
        middle = number;
      }
    }

    return false;
  }
}

File: leetcode/0001-1000/0301-0400/0331-0340/0334-increasing-triplet-subsequence/IncreasingTripletSubsequence.ts
function increasingTriplet(numbers: number[]): boolean {
  let smallest = Number.MAX_VALUE;
  let middle = Number.MAX_VALUE;

  for (const number of numbers) {
    if (number > middle) {
      return true;
    } else if (number < smallest) {
      smallest = number;
    } else {
      middle = number;
    }
  }

  return false;
};

console.log([2, 1, 5, 0, 4, 6]);
